test/unit/data.js

```diff
@@ -874,3 +874,18 @@ test( "Check that the expando is removed when there's no more data", function()
 		}
 	}
 });
+
+test( ".data(prop) does not create expando", function() {
+	expect( 2 );
+
+	var div = jQuery( "<div/>" );
+
+	div.data("foo");
+	equal( false, jQuery.hasData( div[0] ) );
+	// Make sure no expando has been added
+	for ( key in div[ 0 ] ) {
```

Comment:
> This is why we have a code-style

We can codify this, sure, if JSCS can do it. It's just how it is in the whole code base so it's a de-facto standard anyway.